core/deployment/src/main/java/io/quarkus/artemis/core/deployment/ShadowRunTimeConfigs.java

```diff
@@ -0,0 +1,48 @@
+package io.quarkus.artemis.core.deployment;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import io.quarkus.artemis.core.runtime.ArtemisUtil;
+import io.quarkus.runtime.annotations.*;
+
```

Comment:
Can we leave a comment here to mention that it is only used for reading the keys? and it could be improved by https://github.com/quarkusio/quarkus/issues/28415.

Reply:
Yes. I will also try to restructure those classes so it is obvious we access only keys and query presence of values, to furher clarify the use case for those classes.

Reply:
Done. 
@zhfeng care to take another look?

Reply:
yeah, it looks good to me. Thanks again!

Reply:
Hmm, it seems that I have no permission to resolve the conversation. Anyway, please go head for a new release.

Reply:
As soon as the pipeline succeeds (the amount of tests we have, and the fact that all tests also compile natively, means that the build takes quite some time).